src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -116,6 +118,52 @@ class spell_mage_blast_wave : public SpellScriptLoader
         }
 };
 
+// -31641 - Blazing Speed
+class spell_mage_blazing_speed : public SpellScriptLoader
+{
+public:
+    spell_mage_blazing_speed() : SpellScriptLoader("spell_mage_blazing_speed") { }
+
+    class spell_mage_blazing_speed_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_mage_blazing_speed_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_MAGE_BLAZING_SPEED))
+                return false;
+            return true;
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
```

Comment:
This entire check is not needed - script will only run for the intended talent because you are binding it to spell id in spell_script_names table